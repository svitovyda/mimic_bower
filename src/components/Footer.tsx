import * as React from "react";

export const Footer: React.FC = () => {
  return (
    <footer>
      <p>
        <a href="https://github.com/bower/bower.github.io/issues" target="_blank" rel="noreferrer">
          Help improve these docs. Open an issue or pull request.
        </a>
      </p>
      <p>Our supporters: TBA</p>
    </footer>
  );
};
