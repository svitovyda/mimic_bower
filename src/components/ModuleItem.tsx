import type { Package } from "../models/Package";
import * as React from "react";

export interface ModuleItemProp {
  packageElement: Package;
}

export const ModuleItem: React.FC<ModuleItemProp> = ({ packageElement }: ModuleItemProp) => {
  return (
    <div>
      {packageElement.name} {packageElement.owner} {packageElement.stars}
    </div>
  );
};
