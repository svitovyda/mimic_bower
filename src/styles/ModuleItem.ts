import styled from 'styled-components';

export const ItemContainer = styled.div`
  border-bottom: 1px solid #eee;
  height: 35px;
  display: flex;
  align-items: center;
  padding: 10px 0;
`;

const Item = styled.span`
  flex: 1;
`;

export const NameItem = styled(Item)`
  flex: 70;
`;

export const OwnerItem = styled(Item)`
  flex: 15;

  @media only screen and (max-width: 680px) {
    display: none;
  }
`;

export const StarsItem = styled(Item)`
  flex: 15;

  @media only screen and (max-width: 680px) {
    display: none;
  }
`;
