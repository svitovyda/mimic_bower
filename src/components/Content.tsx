import { useContentFetcher } from "../hooks/contentFetcher";
import { ModuleItem } from "./ModuleItem";
import { LoaderAnimation } from "./ui/LoaderAnimation";
import * as React from "react";

export const Content: React.FC = () => {
  const [state, setQuery] = useContentFetcher("");

  React.useEffect(() => {
    setQuery("jquery");
  }, [setQuery]);

  if (state.isLoading) return <LoaderAnimation />;
  if (state.isError) return <div>Some error occurred, try again later!</div>;
  return (
    <div>
      {state.data.map((p) => (
        <ModuleItem key={p.name} packageElement={p} />
      ))}
    </div>
  );
};
