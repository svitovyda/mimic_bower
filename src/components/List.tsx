import type { Package } from '../models/Package';
import { ModuleItem } from './ModuleItem';
import * as React from 'react';

export interface ListProps {
  data: Package[];
}

export const List: React.FC<ListProps> = ({ data }: ListProps) => {
  return (
    <div>
      {data.map((p, i) => (
        <ModuleItem key={`${i}-${p.name}-${p.platform}`} packageElement={p} />
      ))}
    </div>
  );
};
