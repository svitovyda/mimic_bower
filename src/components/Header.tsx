import { HeaderContainer, HeaderLogo, HeaderSubtitle, HeaderTitle } from '../styles/Header';
import * as React from 'react';

export const Header: React.FC = () => {
  return (
    <header>
      <HeaderContainer>
        <HeaderLogo />
        <HeaderTitle>Bower Search</HeaderTitle>
        <HeaderSubtitle>
          Powered by{' '}
          <a href="https://libraries.io/" target="_blank" rel="noreferrer">
            libraries.io
          </a>
        </HeaderSubtitle>
      </HeaderContainer>
    </header>
  );
};
