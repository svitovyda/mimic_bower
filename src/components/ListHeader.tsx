/* eslint-disable jsx-a11y/anchor-is-valid */
import type { SortType } from '../services/ApiServise';
import { ListHeaderContainer, NameLink, OwnerLink, StarsLink } from '../styles/ListHeader';
import * as React from 'react';

export interface ListHeaderProps {
  onSort: (sortBy?: SortType) => void;
}

export const ListHeader: React.FC<ListHeaderProps> = React.memo(({ onSort }: ListHeaderProps) => {
  return (
    <ListHeaderContainer>
      <NameLink onClick={() => onSort()}>Name</NameLink>
      <OwnerLink>Owner</OwnerLink>
      <StarsLink onClick={() => onSort('stars')}>Stars</StarsLink>
    </ListHeaderContainer>
  );
});

ListHeader.displayName = 'ListHeader';
