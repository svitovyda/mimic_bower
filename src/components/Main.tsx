import { Content } from "./Content";
import { Search } from "./Search";
import * as React from "react";

export const Main: React.FC = () => {
  const [query, setQuery] = React.useState<string>("");

  const onNewQuery = React.useCallback(
    (newQuery: string) => {
      setQuery(newQuery);
    },
    [setQuery]
  );

  return (
    <div>
      <Search onQueryChanged={onNewQuery} />
      <Content query={query} />
    </div>
  );
};
