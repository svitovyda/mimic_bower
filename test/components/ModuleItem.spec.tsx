import { ModuleItem } from '../../src/components/ModuleItem';
import type { Package } from '../../src/models/Package';
import { MockList } from '../__mock__/mock';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import * as React from 'react';

describe('ModuleItem', () => {
  it('renders without error', () => {
    const component = render(<ModuleItem packageElement={MockList[0] as Package} />);

    expect(component).toMatchSnapshot();
  });
});
