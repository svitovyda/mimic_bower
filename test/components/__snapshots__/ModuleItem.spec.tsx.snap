// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleItem renders without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  border-bottom: 1px solid #eee;
  height: 35px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 {
  -webkit-flex: 70;
  -ms-flex: 70;
  flex: 70;
}

.c3 {
  -webkit-flex: 15;
  -ms-flex: 15;
  flex: 15;
}

.c4 {
  -webkit-flex: 15;
  -ms-flex: 15;
  flex: 15;
}

@media only screen and (max-width:680px) {
  .c3 {
    display: none;
  }
}

@media only screen and (max-width:680px) {
  .c4 {
    display: none;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <span
          class="c1 c2"
        >
          grunt
        </span>
        <span
          class="c1 c3"
        />
        <span
          class="c1 c4"
        >
          12193
        </span>
      </div>
    </div>
  </body>,
  "container": .c0 {
  border-bottom: 1px solid #eee;
  height: 35px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 {
  -webkit-flex: 70;
  -ms-flex: 70;
  flex: 70;
}

.c3 {
  -webkit-flex: 15;
  -ms-flex: 15;
  flex: 15;
}

.c4 {
  -webkit-flex: 15;
  -ms-flex: 15;
  flex: 15;
}

@media only screen and (max-width:680px) {
  .c3 {
    display: none;
  }
}

@media only screen and (max-width:680px) {
  .c4 {
    display: none;
  }
}

<div>
    <div
      class="c0"
    >
      <span
        class="c1 c2"
      >
        grunt
      </span>
      <span
        class="c1 c3"
      />
      <span
        class="c1 c4"
      >
        12193
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
